features/collection/persistence-osgi/src/main/java/org/opennms/features/collection/persistence/osgi/OsgiPersisterFactory.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.collection.persistence.osgi;
+
+import org.opennms.core.soa.lookup.ServiceLookup;
+import org.opennms.core.soa.lookup.ServiceLookupBuilder;
+import org.opennms.core.soa.support.DefaultServiceRegistry;
+import org.opennms.netmgt.collection.api.Persister;
+import org.opennms.netmgt.collection.api.PersisterFactory;
+import org.opennms.netmgt.collection.api.ServiceParameters;
+import org.opennms.netmgt.rrd.RrdRepository;
+
+public class OsgiPersisterFactory implements PersisterFactory {
+
+    private static final ServiceLookup SERVICE_LOOKUP = new ServiceLookupBuilder(DefaultServiceRegistry.INSTANCE)
+                                                             .blocking().build();
+    private PersisterFactory delegate;
+
+    @Override
+    public Persister createPersister(ServiceParameters params, RrdRepository repository) {
+        Persister persister = createPersister(params, repository, false, false, false);
+        return persister;
+    }
+
+    @Override
+    public Persister createPersister(ServiceParameters params, RrdRepository repository, boolean dontPersistCounters,
+            boolean forceStoreByGroup, boolean dontReorderAttributes) {
+        // There is only one persister that can load from osgi, so use the filter directly
+        delegate = SERVICE_LOOKUP.lookup(PersisterFactory.class, "(strategy=kafka)");
```

Comment:
We should find a way to avoid hard-coding the "kafka" name here since everything else about this persister is already generic.